cmd/config/internal/commands/cmdmigratesetters_test.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package commands_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/cmd/config/internal/commands"
+)
+
+func TestMigrateSettersCommand(t *testing.T) {
+	var tests = []struct {
+		name            string
+		input           string
+		err             string
+		openAPIFile     string
+		expectedOutput  string
+		expectedOpenAPI string
+	}{
+		{
+			name: "migrate-setters-delete-partial-setters",
```

Comment:
@yuwenma Please take a look at migration test cases and feel free to suggest more.

Reply:
One test case comes to my mind is the multi-line partialSetter. Below is the data from a real example, the one that `kpt cfg create-subst` can't parse. :) 

```yaml

spec:
   containers:
   - args:
   - --scoped-namespace=ns1 # {"type":"string","x-kustomize":{"partialSetters" [{"name":"namespace","value":"ns1"}]}}
   - --stderrthreshold=INFO
   - --prometheus-scrape-endpoint=:8888
   command:
   - /configconnector/manager

Reply:
Added the case for list item with `--` prefix. Internal code logic parses the string well. But the issue is with cobra treating it as a flag instead of an input argument when user invokes cli command. We can address that in a different issue.